### PR TITLE
remove all net45 code paths

### LINKED ISSUE #2109: Remove NET45 ifdefs
As of [#2102](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2102), net45 is no longer supported in dev7x. There are ~30 files that still use `#if NET45` that no longer need to.

### PR DESCRIPTION
**Pull request by westin-m**

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs [hunk 1]
```diff
@@ -5,14 +5,6 @@
 using System.Security.Cryptography;
 using Microsoft.IdentityModel.Logging;
 
-#if NET45
-using System.Reflection;
-#endif
-
-#if NET461 || NET462 || NET472 || NETSTANDARD2_0 || NET6_0_OR_GREATER
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:12]: why removing this one?

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs [hunk 2]
```diff
@@ -61,7 +35,6 @@ internal AsymmetricAdapter(SecurityKey key, string algorithm, bool requirePrivat
         {
         }
 
-        // This constructor will be used by NET45 for signing and for RSAKeyWrap
         internal AsymmetricAdapter(SecurityKey key, string algorithm, HashAlgorithm hashAlgorithm, bool requirePrivateKey)
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:38]: Do we need the contructor at all?

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs [hunk 3]
```diff
@@ -395,244 +340,6 @@ private bool VerifyWithRsaCryptoServiceProviderProxyWithLength(byte[] bytes, int
     #endif
 
 #endif
-#endregion
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:398]: which start of region does it relate to? (I probably missed it)
(we can't remove the #endregion without removing the #region )?

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs [hunk 4]
```diff
@@ -395,244 +340,6 @@ private bool VerifyWithRsaCryptoServiceProviderProxyWithLength(byte[] bytes, int
...
-                    };
-
-                    return (bool)verifyDelegate.DynamicInvoke(args);
-                };
-            }
-
-            return _rsaPkcs1VerifyMethod(RSA, input, signature, _lightUpHashAlgorithmName);
-        }
-#endif
 #endregion
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:343]: ah this one needs to be removed (it matches with 
"#region NET45 'lightup' code")

File: src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs [hunk 5]
```diff
@@ -254,12 +223,8 @@ internal static unsafe byte[] UnsafeDecode(string str)
                     fixed (char* src = str)
                     fixed (char* dest = decodedString)
                     {
-#if NET45
-                        for (int index = 0; index < str.Length; index++)
-                            dest[index] = src[index];
-#else
                         Buffer.MemoryCopy(src, dest, str.Length * 2, str.Length * 2);
-#endif
+
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs:227]: extra line?

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs [hunk 6]
```diff
@@ -335,11 +335,6 @@ internal static bool IsSupportedRsaAlgorithm(string algorithm, SecurityKey key)
 
         private static bool IsSupportedRsaPss(SecurityKey key)
         {
-#if NET45
-            // RSA-PSS is not available on .NET 4.5
-            LogHelper.LogInformation(LogMessages.IDX10692);
-            return false;
-#elif NET461 || NET462 || NET472 || NETSTANDARD2_0 || NET6_0_OR_GREATER
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:342]: Do we want to have a `#if` here?
to match with the `#else`, line 358

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs [hunk 7]
```diff
@@ -14,12 +14,7 @@
 using Microsoft.IdentityModel.Json;
 using Microsoft.IdentityModel.Json.Linq;
 
-#if NET452
-using JsonReaderException = Microsoft.IdentityModel.Json.JsonReaderException;
-#else
-using System.Text.Json;
 using JsonReaderException = System.Text.Json.JsonException;
```
Comment (jmprieur) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:17]: this line remains?

File: test/Microsoft.IdentityModel.Tokens.Tests/SignatureProviderTests.cs [hunk 8]
```diff
@@ -414,20 +408,11 @@ public void AsymmetricSignatureProvider_SupportedAlgorithms()
...
-            Module systemCoreModule = typeof(System.Security.Cryptography.AesCryptoServiceProvider).GetTypeInfo().Assembly.GetModules()[0];
-            if (systemCoreModule != null && systemCoreModule.GetType("System.Security.Cryptography.AesCng") != null)
-                return true;
-            return false;
-#else
-            // test for Core
-            return true;
-#endif
-        }
+        //private static bool IsRunningOn462OrGreaterOrCore()
```
Comment (jmprieur) [inline test/Microsoft.IdentityModel.Tokens.Tests/SignatureProviderTests.cs]: do we need this?

**Review by jmprieur — COMMENTED** [review]
Thanks @westin-m
That's a lot of changes !

I asked a few questions for things I found surprising. Otherwise, looks good!

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs (hunk 1 above)
Comment (westin-m) [reply src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:12]: The X509Certificates is an unnecessary using.

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs (hunk 2 above)
Comment (westin-m) [reply src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:38]: Yes, the constructor above it refers to it, and that one is used.

File: src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs (hunk 3 above)
Comment (westin-m) [reply src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs:398]: The diff shows a different endregion being removed; it effectively doesn't matter which one is removed.

File: src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs (hunk 5 above)
Comment (westin-m) [reply src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs:227]: I don't know if it's better style to have an empty line between the memcpy and the other lines. I can remove it.

File: src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs (hunk 6 above)
Comment (westin-m) [reply src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs:342]: This elif seems to cover all of the targets and doesn't feel necessary, but if I'm mistaken then I'm happy to replace it.

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs (hunk 7 above)
Comment (westin-m) [reply test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:17]: System.Text,Json is an unnecessary using.

File: test/Microsoft.IdentityModel.Tokens.Tests/SignatureProviderTests.cs (hunk 8 above)
Comment (westin-m) [reply test/Microsoft.IdentityModel.Tokens.Tests/SignatureProviderTests.cs]: No, I meant to fully delete this.

**Review by jmprieur — APPROVED** [review]

**Pull request merged**